plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/client/ElasticsearchClient.java

```diff
@@ -520,38 +542,135 @@ public boolean indexExists(String index)
+            }
         });
     }
 
+    private IndexMetadata.ObjectType merge(String index, String parentPrefix, IndexMetadata.ObjectType schema1, IndexMetadata.ObjectType schema2)
+    {
+        if (schema2 == null || schema2.getFields().isEmpty()) {
+            return schema1;
+        }
+        if (schema1 == null || schema1.getFields().isEmpty()) {
+            return schema2;
+        }
+        List<IndexMetadata.Field> fields = merge(index, parentPrefix, schema1.getFields(), schema2.getFields());
+        return new IndexMetadata.ObjectType(fields);
+    }
+
+    private List<IndexMetadata.Field> merge(String index, String parentPrefix, List<IndexMetadata.Field> a, List<IndexMetadata.Field> b)
+    {
+        // Non object type fields
+        Set<IndexMetadata.Field> fields = Stream.concat(a.stream(), b.stream())
+                .filter(field -> !(field.getType() instanceof IndexMetadata.ObjectType))
+                .collect(Collectors.toSet());
+
+        // Object Type fields
+        Set<IndexMetadata.Field> objectTypeFields = Stream.concat(a.stream(), b.stream())
+                .filter(field -> field.getType() instanceof IndexMetadata.ObjectType)
+                .collect(Collectors.groupingBy(field -> field.getName()))
+                .entrySet()
+                .stream()
+                .map(nameList -> {
+                    String p = Strings.isNullOrEmpty(parentPrefix) ? nameList.getKey() : getFlattenedKey(parentPrefix, nameList.getKey());
+                    IndexMetadata.ObjectType type = nameList.getValue().stream()
+                            .map(field -> (IndexMetadata.ObjectType) field.getType())
+                            .reduce((nestedSchema1, nestedSchema2) -> merge(index, p, nestedSchema1, nestedSchema2))
+                            .get();
+                    IndexMetadata.Field f = nameList.getValue().iterator().next();
+                    return new IndexMetadata.Field(f.isArray(), f.getName(), type);
+                })
+                .collect(Collectors.toSet());
+
+        fields.addAll(objectTypeFields);
+        fields = checkAndCorrectMismatchFields(index, parentPrefix, fields);
+
+        return new ArrayList<>(fields);
+    }
+
+    private String getFlattenedKey(String parent, String child)
+    {
+        return format("%s.%s", parent, child);
+    }
+
+    private Set<IndexMetadata.Field> checkAndCorrectMismatchFields(String index, String parent, Set<IndexMetadata.Field> fields)
+    {
+        Map<String, List<IndexMetadata.Field>> fieldsMap = fields.stream()
+                .collect(Collectors.groupingBy(IndexMetadata.Field::getName));
+
+        Map<String, List<IndexMetadata.Field>> typeMismatchFields = fieldsMap.entrySet()
+                .stream()
+                .filter(fieldNameToFieldsList -> fieldNameToFieldsList.getValue().size() > 1)
+                .collect(Collectors.toMap(o -> o.getKey(), o -> o.getValue()));
```

Comment:
Use `Function.identity()` here for both key/value functions.

Reply:
Also if you aren't changing the map in future and won't have null values use the `toImmutableMap` collector instead of `toMap`.

Reply:
`Function.identity()` can't be applied on this I think. Because it returns a `Map.Entry`.
Instead we could do `.collect(toImmutableMap(Map.Entry::getKey, Map.Entry::getValue));`

Please help me out how to do this using `Function.identity()` if possible.

Reply:
I thought this should be possible doing
`Collectors.toMap(Function.identity(), Function.identity())` but honestly the way you suggested works fine too :)